### DISCUSSION TITLE
svelte checkbox Row Selection all example?

### DISCUSSION THREAD

**Discussion by exqmjmz** [TanStack/table#4770] [Q&A] [answered]
I implemented head and cell output html checkbox
![image](https://user-images.githubusercontent.com/12406935/227163115-4e5c68ef-3a62-45d7-8b16-96df8992880a.png)

I refer to React Row Selection, but svelte doesn't know what code to implement.
![image](https://user-images.githubusercontent.com/12406935/227163919-3d16fab5-2f0e-4d57-bbeb-3d5b3981de36.png)
![image](https://user-images.githubusercontent.com/12406935/227163940-7385a1a5-5492-4c55-88b0-f2f6d75e9435.png)
![image](https://user-images.githubusercontent.com/12406935/227164121-a80098a6-d3c3-4052-a21c-70ef012a2dcc.png)

**Comment by ollydowning** [comment, marked as answer]
Hi @ljm000,

I was struggling with this myself, and have come up with the following solution.

First, create a `RowCheckBox.svelte` component:

``` Svelte
<script lang="ts">
    export let checked, indeterminate, onChange;
</script>

<input class="checkbox" type="checkbox" on:change={onChange} {checked} bind:indeterminate={indeterminate}/>
```

Next, define your column's as normal, passing in the following items we'll need in the component.

``` TypeScript
const defaultColumns: ColumnDef<Child>[] = [
    {
        id: 'select',
        header: ({table}) => renderComponent(RowCheckBox, {
            checked: table.getIsAllRowsSelected(),
            indeterminate: table.getIsSomeRowsSelected(),
            onChange: table.getToggleAllRowsSelectedHandler(),
        }),
        cell: (props) => renderComponent(RowCheckBox, {
            checked: props.row.getIsSelected(),
            disabled: !props.row.getCanSelect(),
            indeterminate: props.row.getIsSomeSelected(),
            onChange: props.row.getToggleSelectedHandler(),
        })

    },
    // Rest of Columns
]
```

Before we create our table, we need to handle ticking a checkbox by defining a state object and a function to handle updates. This is indirectly passed along to the component, through TanStack via `on:change={onChange}`.

``` TypeScript
let selection: RowSelectionState = {};

const onSelect = (updater: any) => {
    if (updater instanceof Function) {
        selection = updater(selection);
    } else {
        selection = updater;
    }

    options.update((old) => ({
        ...old,
        state: {
            ...old.state,
            rowSelection: selection,
        },
    } as TableOptions<Child>));
}
```

Finally this can all be passed into table options:

``` TypeScript
const options = writable<TableOptions<Child>>({
    //... other options
    columns: defaultColumns,
    state: {
        //... other state
        rowSelection: selection,
    },
    enableRowSelection: true,
    onRowSelectionChange: onSelect,
    //... other options
})
```

Let me know how you get on :)

**Comment by c-kirkeby** [comment]
Any idea how to use the `on:change` handlers from Svelte as part of the `cell` definition? Seems odd to have to define an `onChange` prop to proxy the underlying handler.
